main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -473,6 +474,129 @@ object EntryPoints {
 
   }
 
+  private object WrapInDefaultHandlers {
+    private case class HandlerInfo(
+                                    handlerSym: Symbol.DefnSym,
+                                    handlerDef: TypedAst.Def,
+                                    handlerDefSymUse: SymUse.DefSymUse,
+                                    handledEff: Type,
+                                    handledSym: Symbol.EffSym,
+                                    generatedPrimitiveEffs: List[Type]
+                                  )
+
+    /**
+      * WrapInDefaultHandlers takes a root and returns a new root where the entry points' definitions
+      * have been wrapped in a default handler for any extra non-primitive effects
+      * (eg. Assert)
+      *
+      * Takes
+      * {{{
+      * def f(arg1: tpe1, ...): tpe \ ef = exp
+      * }}}
+      * Returns
+      * {{{
+      * def f(arg1: tpe1, ...): tpe \ efHandlers + (ef - efsDefault) = efDefault1.handle(_ -> efDefault2.handle(... -> exp))
```

Comment:
Here efHandlers is in plural. Maybe the example would be better with a single handler and then you can explain that it deals with multiple handlers.